Charts/README_CHARTS.md

```diff
@@ -0,0 +1,71 @@
+
+This standalone Helm related workflow is independent of the rest of the workflows except that the _pypi workflow has _helm added to its `needs` in `ci.yml`, making sure we only publish to pypi with valid SemVer tags.
+
+TODO: for a project that publishes containers referred to by the helm chart, the `_containers.yml` workflow should be added to the `needs` of _helm in `ci.yml`.
+
+## Schema Generation
+
+Schema generation for charts' `values.yaml` is handled by [helm-values-schema-json](https://github.com/losisin/helm-values-schema-json). Which is in turn controlled by annotations in the default [values.yaml](fastcs/values.yaml) file.
+
+The generated schema file will be called `values.schema.json` and will be placed in the same directory as the `values.yaml` file and commited to the repo. This is done automatically by a [pre-commit hook](https://github.com/DiamondLightSource/FastCS/blob/8232393b38cc8e0eee00680e95c2ce06e7983ba6/.pre-commit-config.yaml#L27-L33). Therefore, when developing charts you can update schemas with:
+
+```bash
+git add . ; pre-commit
+```
+
+Note that this standard name for the schema file means that it is packaged up with the helm chart and available for schema checks in ArgoCD for example.
+
+## schemas folder
+
+The schemas folder allows us to declare the schemas you want to publish to the release.
+
+It should contain:
+
+- A symlink to each of the `values.schema.json` files in the subfolders of `Charts`. The symlink should have a unique name, e.g. `fastcs-values.schema.json`, thus allowing multiple schemas to be published per repo.
+- A service schema file which references the above and can be used to validate `values.yaml` in epics-containers services repos, where the these charts will be used as sub-charts. e.g. [fastcs-service.schema.json](../schemas/fastcs-service.schema.json)
+
+The service schema files are hand coded as they are extremely simple. The symlinks are also manually created at present. (both of these could potentially be automated).
+
+## Debuging/Development In-Cluster
+
+The `fastcs` helm chart has two variables to enable debugging/development in-cluster:
+
+- `editable`: When true:
+  - a PVC is created.
+  - The debug version of the container image is referenced.
+  - The contents of /workspaces and /venv are copied into the PVC.
+  - The venv from the debug image is an editable install of the project source code in /workspaces.
+  - The PVC folders are mounted over the corresponding folders in the container.
+- `autostart`:
+  - When false, the container starts with PID 1 as sleep infinity.
+  - When true, the container starts with its normal entrypoint.
+
+In combination these flags can be used to debug or develop in-cluster.
+
+An initial demonstration script to use these features is provided in [debug.py](https://github.com/epics-containers/p47-services/blob/add-fastcs/debug.py) in the `p47-services` repo.
```

Comment:
If this is fastcs specific should we just put in in here as the reference version for now? Could it be made generic for all services by moving autostart and editable to globals?